### ISSUE TITLE
IntegrationError on latest 1.2.0 update

### ISSUE THREAD

**Issue by morgoth** [stripe/react-stripe-js#152]
With latest update of react-stripe-js (1.1.2 => 1.2.0), I'm getting error: `IntegrationError: Can only create one Element of type card`.

My code looks as follow:
```
      <section>
        <div>
          {paymentMethod === 'card' && (
              <CardElement
                onReady={() => setIsPaymentElementCompleted(false)}
                onChange={onPaymentElementChange}
                className="form-control form-control--sm"
                options={CARD_OPTIONS}
              />
          )}
          {paymentMethod === 'sepa' && (
              <IbanElement
                onReady={() => setIsPaymentElementCompleted(false)}
                onChange={onPaymentElementChange}
                className="form-control form-control--sm"
                options={IBAN_OPTIONS}
              />
          )}
        </div>
      </section>
```

There is an UI element that can switch between those 2 methods (via `paymentMehtod`), so CardElement and IbanElement will be indeed created again, however this was working fine before.

Is this a regression?

**Comment by christopher-stripe** [comment]
Did you happen to upgrade React from 16 to 17 at the same time? React 17 changed the timing of when a `useEffect` cleanup function is run. We use a cleanup function to call [element.destroy](https://stripe.com/docs/js/element/other_methods/destroy), which looks relevant here.

**Comment by morgoth** [comment]
No, I was using React 17 before, so it worked fine before on React 17

**Comment by christopher-stripe** [comment]
I published a fix for this in `v1.2.1`. Thanks for letting us know!

**Comment by morgoth** [comment]
Thank you for the quick fix. I confirm it works fine now.
